socorro/tests/external/es/test_crashstorage.py

```diff
@@ -378,16 +333,111 @@ def test_index_data_capture(self):
             mm.assert_histogram_once("processor.es.index", tags=["outcome:successful"])
             mm.assert_histogram_once("processor.es.index", tags=["outcome:failed"])
 
+    def test_delete_expired_indices(self, es_helper):
+        # Delete any existing indices first
+        es_helper.delete_indices()
+        es_helper.refresh()
+        es_helper.health_check()
+
+        # Create an index > retention_policy
+        crashstorage = self.build_crashstorage()
+        template = crashstorage.get_index_template()
+        now = utc_now()
+        current_index_name = now.strftime(template)
+        before_retention_policy = now - timedelta(weeks=crashstorage.retention_policy)
+        old_index_name = before_retention_policy.strftime(template)
+
+        crashstorage.create_index(current_index_name)
+        crashstorage.create_index(old_index_name)
+        es_helper.health_check()
+        assert list(es_helper.get_indices()) == [old_index_name, current_index_name]
+
+        # Delete all expired indices and verify old one is gone and new one is still
+        # there
+        crashstorage.delete_expired_indices()
+        es_helper.health_check()
+        assert list(es_helper.get_indices()) == [current_index_name]
+
+    # NOTE(willkg): This is dependent on supersearch fields. As we adjust
+    # supersearch fields, we may need to update this test.
+    @pytest.mark.parametrize(
+        "key, value, expected_value",
+        [
+            # Non-string keyword converted to "BAD DATA"
+            ("processed_crash.background_task_name", 44, "BAD DATA"),
+            # Long keywords are truncated
+            pytest.param(
+                "processed_crash.background_task_name",
+                "a" * 10_001,
+                "a" * 10_000,
+                id="long_keywords_truncated",
+            ),
+            # Non-string string converted to "BAD DATA"
+            ("processed_crash.user_comments", 44, "BAD DATA"),
+            # Long strings are truncated
+            pytest.param(
+                "processed_crash.user_comments",
+                "a" * 32_767,
+                "a" * 32_766,
+                id="long_strings_truncated",
+            ),
+            # Out-of-bounds integers are removed
+            (
+                "processed_crash.mac_available_memory_sysctl",
+                2_147_483_999,
+                REMOVED_VALUE,
+            ),
+            # Out-of-bounds longs are removed
+            (
+                "processed_crash.memory_explicit",
+                9_223_372_036_854_775_999,
+                REMOVED_VALUE,
+            ),
+            # Booleans are converted
+            # FIXME(willkg): fix_boolean is never called--that's wrong
```

Comment:
When writing this, I discovered `fix_boolean` is never called. Fixing that is a new bug because it may have consequences on indexing and searching across indexes. I'll look into it later.